### DISCUSSION TITLE
Question about Porous Media - POROUS_VOLUME_FRACTION does not appear to do anything

### DISCUSSION THREAD

**Discussion by uporersith** [firemodels/fds#12226] [Q&A] [answered]
I am testing the porous media feature and use the porous_media.fds sample file, however I find that the POROUS_VOLUME_FRACTION value does not appear to affect the results. Looking at the code in part.f90, I notice that LPC%LENGTH is calculated, however it is not used anywhere in the code. I expected it to be used in the POROUS_DRAG case(line 2724), to match the equation 15.6 (the little delta) but I don't see it in the code, maybe I missed it.  should SFAC have the LPC%LENGTH in it? 

`CASE (POROUS_DRAG)

         TMP_G  = MAX(TMPMIN,TMP(IIG_OLD,JJG_OLD,KKG_OLD))
         ZZ_GET(1:N_TRACKED_SPECIES) = ZZ(IIG_OLD,JJG_OLD,KKG_OLD,1:N_TRACKED_SPECIES)
         CALL GET_VISCOSITY(ZZ_GET,MU_FILM,TMP_G)
         Y_TERM = LPC%DRAG_COEFFICIENT * RHO_G /SQRT(LPC%PERMEABILITY)*QREL
         K_TERM = MU_FILM/LPC%PERMEABILITY
         SFAC = 1._EB/RHO_G
         ACCEL_X = -(MIN(DX(IIG_OLD),LP%DX)/DX(IIG_OLD))*(K_TERM(1)+Y_TERM(1))*UBAR*SFAC
         ACCEL_Y = -(MIN(DY(JJG_OLD),LP%DY)/DY(JJG_OLD))*(K_TERM(2)+Y_TERM(2))*VBAR*SFAC
         ACCEL_Z = -(MIN(DZ(KKG_OLD),LP%DZ)/DZ(KKG_OLD))*(K_TERM(3)+Y_TERM(3))*WBAR*SFAC
         DRAG_MAX(1) = ACCEL_X/(-UBAR-SIGN(1._EB,UBAR)*TWO_EPSILON_EB)
         DRAG_MAX(2) = ACCEL_Y/(-VBAR-SIGN(1._EB,VBAR)*TWO_EPSILON_EB)
         DRAG_MAX(3) = ACCEL_Z/(-WBAR-SIGN(1._EB,WBAR)*TWO_EPSILON_EB)
   END SELECT
`

**Comment by mcgratta** [comment]
```
      CASE (POROUS_DRAG)

         MPUA = 0._EB
         LP_VOLUME = LPC%POROUS_VOLUME_FRACTION*LP%DX*LP%DY*LP%DZ
         SELECT CASE (LP_SF%GEOMETRY)
            CASE (SURF_CARTESIAN)
               B1%AREA = LP_VOLUME/LP_SF%THICKNESS
               LPC%LENGTH = SQRT(B1%AREA )
               IF (LP_SF%THERMAL_BC_INDEX==THERMALLY_THICK) THEN
                  DO N=1,LP_SF%N_LAYERS
                     LP%MASS = LP%MASS + LP_SF%LAYER_THICKNESS(N)*LP_SF%LAYER_DENSITY(N)
                  END DO
                  LP%MASS = LP%MASS*B1%AREA
                  LP%RADIUS = SUM(LP_SF%LAYER_THICKNESS(:))/2._EB
               ENDIF
```

**Comment by mcgratta** [comment]
This bit of code is in the set up routine called `INITIALIZE_SINGLE_PARTICLE`

**Reply by uporersith** [reply]
I see how LPC%LENGTH is calculated, but I don't understand how it is used in the drag calculation. Or how is the delta in equation 15.6 in the user's manual is implemented in the code. I only ask this because it didn't seem like the results were changing when i varied POROUS_VOLUME_FRACTION.

**Comment by mcgratta** [comment]
Here is my go-to command
```
[mcgratta@burn Source]$ grep POROUS_VOLUME_FRACTION *f90
part.f90:         LP_VOLUME = LPC%POROUS_VOLUME_FRACTION*LP%DX*LP%DY*LP%DZ
read.f90:REAL(EB) :: DRAG_COEFFICIENT(3),FREE_AREA_FRACTION,PERMEABILITY(3),POROUS_VOLUME_FRACTION,SHAPE_FACTOR
read.f90:                N_STRATA,NEW_PARTICLE_INCREMENT,ORIENTATION,PERMEABILITY,POROUS_VOLUME_FRACTION,&
read.f90:   LPC%POROUS_VOLUME_FRACTION = POROUS_VOLUME_FRACTION
read.f90:POROUS_VOLUME_FRACTION = -1._EB
read.f90:            LPC%POROUS_VOLUME_FRACTION = IN%MASS_PER_VOLUME / LPC%DENSITY
read.f90:      IF (LPC%DRAG_LAW == POROUS_DRAG .AND. LPC%POROUS_VOLUME_FRACTION <= 0._EB) THEN
read.f90:                                  ' requires a PART POROUS_VOLUME_FRACTION or INIT MASS_PER_VOLUME'
type.f90:   REAL(EB) :: POROUS_VOLUME_FRACTION     !< Volume fraction of cell open to flow in porous media model
```

**Comment by mcgratta** [comment]
That delta is the size of the grid cell that is assumed to be filled with the porous media. You don't see it explicitly in the coded formula, but it appears implicitly when the drag term is added to the momentum equation.

**Comment by uporersith** [comment]
I'm attaching a simple test case. 
I'm expecting the pres_1 to be different than pres_2 because they have different POROUS_VOLUME_FRACTION.  but instead, I get the same values for pres_1 and pres_2 no matter what I change the POROUS_VOLUME_FRACTION to. In fact if I set the POROUS_VOLUME_FRACTION to be the same for both cases I get the same values(0.9 and 1.01).

So I the question is, what does POROUS_VOLUME_FRACTION actually do in the calculation? it does not appear in equation 15.6. 

[porous_media.txt](https://github.com/firemodels/fds/files/13342983/porous_media.txt)

**Comment by drjfloyd** [comment, marked as answer]
The POROUS_VOLUME_FRACTION does not define the drag. That is used along with the SURF dimension to determine the surface area for heat transfer.

**Reply by uporersith** [reply]
Thank you Dr. Floyd. That clears it up. Section 15.4.8 in the user's manual confused me. I got the impression that all the parameters in that section were used in computing the drag.
